### DISCUSSION TITLE
rdChemReactions from SMARTS without atom mappings

### DISCUSSION THREAD

**Discussion by MartinHoffmannJena** [rdkit/rdkit#8047] [Q&A] [answered]
Hi,

My scenario looks like this:

I have a molecule "CCCCCCO" and the following reaction SMARTS (without atom mappings): "[#6]-[#6]-[#6]-[#8]>>[#6]-[#6]-[#6]-[#28]"

`from rdkit import Chem`
`from rdkit.Chem import rdChemReactions`
`mol1 = "Chem.MolFromSmiles("CCCCCCO")`
`rxn = rdChemReactions.ReactionFromSmarts("[#6]-[#6]-[#6]-[#8]>>[#6]-[#6]-[#6]-[#28]") `
`print(Chem.MolToSmiles(rxn.RunReactant((mol1),0)[0][0]))`

The expected outcome would be, that the product is "CCCCCC[Ni]", the actual output however is "CCC[Ni"].

If I add atom mappings to the reaction smarts the expected output is produced, but I dont fully understand why atom mappings are necessary here.

**Comment by bp-kelley** [comment, marked as answer]
The reason is simple, you have six carbons in the reactant and three in the
product, which ones from the reagent are in the product?

There is no way for the reaction to know which carbon are mapped, so it
makes no assumptions.

**Reply by MartinHoffmannJena** [reply]
Thank you for your response.

From what I understand, this reaction says "a chain of 3 carbons and 1 oxygen get transformed into a chain of 3 carbons and 1 Ni". My reactant only has one possible chain of 3 carbons and 1 oxygen, which is the latter three carbons and the one oxygen in there. The first three carbons can never be part of a CCCO pattern match, so isnt it unambiguous which carbons are are referenced here?

**Reply by bp-kelley** [reply]
It is an assumption that the oxygen is changing in to a Ni.

**Reply by bp-kelley** [reply]
Or that the carbons in the product come from the reagent for that matter
